source/game_sa/Tasks/TaskTypes/TaskComplexDestroyCarMelee.cpp

```diff
@@ -0,0 +1,180 @@
+    case TASK_SIMPLE_FIGHT_CTRL: {
+        if (ped->bStayInSamePlace) { // Inverted
+            const auto finished = m_nTimeMs != -1 && CTimer::GetTimeInMS() - m_nTimeMs > 3000;
+            return CreateSubTask(finished ? TASK_FINISHED : TASK_SIMPLE_PAUSE, ped);
+        }
+        CalculateSearchPositionAndRanges(ped);
+        return CreateSubTask(TASK_COMPLEX_SEEK_ENTITY, ped);
+    }
+    case TASK_COMPLEX_GO_TO_POINT_AND_STAND_STILL: {
+        CalculateSearchPositionAndRanges(ped);
+        if (IsPointInSphere(ped->GetPosition(), m_VehToDestroy->GetPosition(), m_MaxTargetFightDist)) {
+            return CreateSubTask(TASK_SIMPLE_FIGHT_CTRL, ped);
+        }
+        return CreateSubTask(ped->bStayInSamePlace ? TASK_SIMPLE_PAUSE : TASK_COMPLEX_SEEK_ENTITY, ped);
+    }
+    case TASK_COMPLEX_SEEK_ENTITY: {
+        return CreateSubTask(TASK_SIMPLE_FIGHT_CTRL, ped);
+    }
+    default:
+        return nullptr;
+    }
+}
+
+// 0x62DB20
+CTask* CTaskComplexDestroyCarMelee::CreateFirstSubTask(CPed* ped) {
+    byteD = false;
+
+    // CWeaponInfo::GetWeaponInfo(ped->GetActiveWeapon().m_nType); // unused
+
+    CalculateSearchPositionAndRanges(ped);
+
+    const auto& pedPos = ped->GetPosition();
+
+    if (IsPointInSphere(pedPos, m_VehToDestroy->GetPosition(), m_MaxFightCtrlRadius)) {
+        return CreateSubTask(ped->bStayInSamePlace ? TASK_SIMPLE_PAUSE : TASK_COMPLEX_SEEK_ENTITY, ped);
+    }
+
+    ped->m_fAimingRotation = CGeneral::GetRadianAngleBetweenPoints(m_VehiclePos, pedPos);
+    return CreateSubTask(TASK_SIMPLE_FIGHT_CTRL, ped);
+}
+
+// 0x62DDB0
+CTask* CTaskComplexDestroyCarMelee::ControlSubTask(CPed* ped) {
+    if (byteD) {
+        return CreateFirstSubTask(ped);
+    }
+
+    if (m_VehToDestroy->m_fHealth <= 0.0f)
+        return nullptr;
+
+    switch (m_pSubTask->GetTaskType()) {
+    case TASK_COMPLEX_SEEK_ENTITY:
+    case TASK_COMPLEX_GO_TO_POINT_AND_STAND_STILL: {
+        CalculateSearchPositionAndRanges(ped);
+        if (DistanceBetweenPoints(ped->GetPosition(), m_VehToDestroy->GetPosition()) > sq(m_MaxTargetFightDist)) {
+            return CreateSubTask(TASK_SIMPLE_FIGHT_CTRL, ped);
+        }
+        return m_pSubTask;
+    }
+    case TASK_SIMPLE_FIGHT_CTRL:
+        CalculateSearchPositionAndRanges(ped);
+        if (DistanceBetweenPoints(ped->GetPosition(), m_VehToDestroy->GetPosition()) > m_MaxTargetFightDist + 0.6f) {
+            return CreateSubTask(TASK_COMPLEX_SEEK_ENTITY, ped);
+        }
+        // todo:
+        return m_pSubTask;
+    default:
+        return nullptr;
+    }
+}
+
+// 0x6289F0
+void CTaskComplexDestroyCarMelee::CalculateSearchPositionAndRanges(CPed* ped) {
+    m_MaxTargetFightDist = m_MaxFightCtrlRadius = m_VehToDestroy->GetModelInfo()->GetColModel()->GetBoundRadius() + 0.35f;
+
+    const auto dir = CVector2D{ ped->GetPosition() - m_VehToDestroy->GetPosition() };
+    m_PedVehicleAngleRad = std::atan2(dir.x, -dir.y); // Or `std::atan2(dir.y, dir.x)` should work too (?)
```

Comment:
```suggestion
    const auto dir = CVector2D{ m_VehToDestroy->GetPosition() - ped->GetPosition() };
    m_PedVehicleAngleRad = dir.Heading();
```
yours: `atan2((a - b).x, -(a - b).y)`
mine: `atan2(-(b - a).x, (b - a).y)`

Reply:
or using 3D vector is fine too:
```suggestion
    m_PedVehicleAngleRad = (ped->GetPosition() - m_VehToDestroy->GetPosition()).Heading();
```

Reply:
Oh lol I didn't know there's a function for that, neat.

Reply:
or using 3D vector is fine too:
```suggestion
    m_PedVehicleAngleRad = (ped->GetPosition() - m_VehToDestroy->GetPosition()).Heading();
```

Reply:
the subtraction must be inverted
![Capture](https://user-images.githubusercontent.com/65975406/187084571-04a89f29-90ef-4dda-82dd-ac9daa90ca65.PNG)
^second atan is from Heading() func

Reply:
Right!
I'm lazy to test it, r u sure it'll work as expected? 😂😂

Reply:
like if it won't then it's the algebra's problem 😀

Reply:
😂😂

Reply:
or using 3D vector is fine too:
```suggestion
    m_PedVehicleAngleRad = (m_VehToDestroy->GetPosition() - ped->GetPosition()).Heading();
```